### PR TITLE
aixPB: xlc13 changes

### PR DESCRIPTION
**Pull request by aixtools**
##### Checklist

- [x] commit message has one of the [standard prefixes](https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/master/FAQ.md#commit-messages)

### CODE REVIEW

File: ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml [hunk 1]
```diff
@@ -13,67 +13,85 @@
...
+      when: result.xlc is defined
+      tags: xlc13
+
+    - name: Clean XLC13 tmp files
+      file:
+        path: /tmp/usr
+        state: absent
+      tags: xlc13
+      when: xlc13_images is defined
+  when: xlc13.stat.islnk is undefined
```
Comment (Haroon-Khel) [inline ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml:48]: I think this should be `when: xlc13.stat.islnk is _not_ undefined`, as this `when` condition covers all the tasks which install xlc13

**Review by Haroon-Khel — CHANGES_REQUESTED** [review]

File: ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml [hunk 2]
```diff
@@ -13,67 +13,85 @@
...
-    src: /Vendor_Files/aix/XLC/IBM_XL_C_C___FOR_AIX_V13.1.3_EMG.tar.gz
-    dest: /tmp
-    remote_src: no
-  when: xlc13.stat.islnk is not defined
-  tags: xlc13
+- name: Check for Vendor Files
+  block:
+    - name: Check for Vendor Files
+      stat:
+        path: /Vendor_Files/aix/XLC/IBM_XL_C_C___FOR_AIX_V13.1.3_EMG.tar.gz
```
Comment (Haroon-Khel) [inline ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml:20]: This task checks for /Vendor_Files/aix/XLC/IBM_XL_C_C___FOR_AIX_V13.1.3_EMG.tar.gz on the target machine, where as it should check on the local machine. This can be done by adding `delegate_to: localhost`, but when doing that I get the following error

`fatal: [p9-aix1-ojdk06.osuosl.org]: FAILED! => {"changed": false, "module_stderr": "sudo: a password is required\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}`

Which makes me think that this particular task is not needed; I would prefer that the folllowing task throws an error if that file isnt present on the local machine, instead of skipped tasks, as a way of notifying the user

File: ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml [hunk 3]
```diff
@@ -13,67 +13,85 @@
...
+#   file: src=/opt/IBM/xlC/13.1.3/bin/xlc_r dest=/usr/bin/xlC_r state=link
+#   when: xlc13.stat.islnk is not defined
+#   tags: xlc13
 
-- name: Clean XLC13 tmp files
-  file:
-    path: /tmp/usr
-    state: absent
-  tags: xlc13
+# - name: Symlink - /usr/bin/gxlC
```
Comment (Haroon-Khel) [inline ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml:94]: Out of curiosity, why are these symlink tasks omitted? I would assume these get used somewhere by a process, but I am not entirely sure.

Comment (aixtools) [reply ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml:94]: It makes them the default. IMHO: better practice will be to set PATH for the compiler wanted (either in a .profile, but better in the script, or as a symlink in the $HOME/bin.

Remember - we are also installing the v16 compiler. Why does that not get symlinks?

So, I left them here - to have this discussion :)

File: ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml (hunk 2 above)
Comment (aixtools) [reply ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml:20]: What I have not done yet - but am planning to do is use a set_fact: statement to say - not ready at the end.

I feel it is important to do as much as possible - where/when not being there does not impact the rest of the playbook.

Nowhere is there documentation on what this tar file is, where it comes from. imho - it is a hack for a previous/prior location. So, I do not want to delete it because it is provider/location bound - but I don't want the playbook stopping in it's tracks.

IMHO: better to say (not ready, need xlc13) (so the not_ready fact needs to become a list of messages that get printed at the end.

As to locations: I want, once I am farther along, to have a template file of variables that can be set for where sources are coming from (e.g, vendor files, nimclient, etc..). But these are issues that come after the core logic runs end to end - and reports on system readiness. Idempotent!

File: ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml (hunk 1 above)
Comment (aixtools) [reply ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml:48]: I am still learning syntax. Had a big fight with the linter over block syntax. As things move I leave the double when: - and you may also notice that I did it differently in the xlcv16 - as I had more experience.

So, heartfelt thx for the feedback.

File: ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml (hunk 2 above)
Comment (aixtools) [reply ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml:20]: further, I read the task to mean the vendor files are on the target, not on the localhost (i.e., ansible control node).

Since the compiler is not installed - check if vendor files are there - opens block.
within the block - there are additional checks
when the tar file exists, unarchive it - register that we have install(p) images.

since we have images (in xlc16 i discovered the condition skipped which I presonally perfer - so the command: installp ... will become when _images is not skipped and rc==0

Also, a register value with a . (dot) in it - as the original has - is confusing imho - makes it look like a complex variable. So I have a later change with a new variable name as well.

In short, there are multiple conditions that need to be met within the block - not just the initial condition. That block also applies that condition is just background behavior irrelevant to the tasks within the block.

File: ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml (hunk 3 above)
Comment (aixtools) [reply ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml:94]: note also the comment: fail: or debug:

If you really want a playbook to stop dead in it's track - change debug: to fail: to be explicit about it.

Again, my preference will be to set a fact that summarizes what 'failed' aka is missing (the xlcv13 compiler) - and at the end of the playbook have a summary of what is needed to be fully prepared.

I really do not like stop and go (e.g., I use `make -i` to get an overview of what is broken rather than trial and trial and trial).

Comment (sxa) [reply ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml:94]: > It makes them the default. IMHO: better practice will be to set PATH

I can understand that but there's no way I can accept this unless we are 100% sure that it wouldn't cause a build failure at the moment if the playbook was run on a new machine which didn't have the symlinks.

**Review by sxa — CHANGES_REQUESTED** [review]

**Comment by aixtools** [comment]
As it has been some months - and I did not understand where VendorFiles were found (I thought on the target, rather than management node) - I shall restart - probably with a forced push after resetting to the current situation.

File: ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml (hunk 3 above)
Comment (aixtools) [reply ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml:94]: I only commented them away - asif removing them. i.e., for existing situation - nothing changes, existing symlinks are not removed - AND - `missing` links are not placed so that users need to add something to PATH.

FYI: the original src: values are (mostly, if not all) symlinks themselves.

**Comment by aixtools** [comment]
FYI: when XLC is not installed and Vendor Files are missing - it looks like this:
```
aixtools@nim.bak:[/home/aixtools/openjdk-infrastructure/ansible/playbooks]ansible-playbook --tags  "xlc13" -i ~/osuosl/inventory.ojdk06 ./*AIX*/main.yml

PLAY [all] ************************************************************************************************************************************************

TASK [Gathering Facts] ************************************************************************************************************************************
[WARNING]: Platform aix on host ojdk06 is using the discovered Python interpreter at /usr/bin/python, but future installation of another Python
interpreter could change the meaning of that path. See https://docs.ansible.com/ansible/2.10/reference_appendices/interpreter_discovery.html for more
information.
ok: [ojdk06]

TASK [xlc_v13 : Checking for XLC13 availability] **********************************************************************************************************
ok: [ojdk06]

TASK [xlc_v13 : debug] ************************************************************************************************************************************
skipping: [ojdk06]

TASK [xlc_v13 : Transfer and Extract XLC13] ***************************************************************************************************************
fatal: [ojdk06]: FAILED! => {"changed": false, "msg": "Could not find or access '/Vendor_Files/aix/XLC/IBM_XL_C_C___FOR_AIX_V13.1.3_EMG.tar.gz' on the Ansible Controller.\nIf you are using a module and expect the file to exist on the remote, see the remote_src option"}
...ignoring

TASK [xlc_v13 : Install IBM XLC13 - installp] *************************************************************************************************************
skipping: [ojdk06]

TASK [xlc_v13 : Remind that installp error messages may be ignored] ***************************************************************************************
skipping: [ojdk06]

TASK [xlc_v13 : Clean XLC13 tmp files] ********************************************************************************************************************
ok: [ojdk06]

TASK [xlc_v13 : TestIBM XLC13] ****************************************************************************************************************************
fatal: [ojdk06]: FAILED! => {"changed": false, "cmd": "/opt/IBM/xlC/13.1.3/bin/xlc -qversion", "msg": "[Errno 2] No such file or directory", "rc": 2}
...ignoring

TASK [xlc_v13 : Display XLC13 qversion information] *******************************************************************************************************
skipping: [ojdk06]

TASK [xlc_v13 : Inform about XLC13 not found] *************************************************************************************************************
ok: [ojdk06] => {
    "msg": "/opt/IBM/xlC/13.1.3/bin/xlc -qversion returns: [Errno 2] No such file or directory"
}

PLAY RECAP ************************************************************************************************************************************************
ojdk06                     : ok=6    changed=0    unreachable=0    failed=0    skipped=4    rescued=0    ignored=2
```

File: ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml (hunk 1 above)
Comment (aixtools) [reply ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml]: The new code should do the trick - but need your verification using vendor files - as I do not have any.

**Comment by Haroon-Khel** [comment]
Having tested this on ojdk06, the changes run fine, both when xlc13 is and isnt installed. Now need to test a build

**Review by Haroon-Khel — COMMENTED** [review]
Regarding omitting the symlinks; /opt/IBM/xlC/13.1.3/bin is in the [$PATH variable](https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/e56d94bac26cb0fbe9beafaae88f2245fd0431a9/ansible/playbooks/AdoptOpenJDK_AIX_Playbook/main.yml#L121) before /usr/bin, so its likely that the symlinks are not needed.

Im testing this on build-osuosl-aix71-ppc64-1. Im going to temporarily remove the symlinks and run a few builds and tests on it

**Comment by Haroon-Khel** [comment]
build-osuosl-aix71-ppc64-1 didnt even have four out of the six symlinks 

```
/usr/bin/xlc++
/usr/bin/xlc_r
/usr/bin/xlC_r
/usr/bin/gxlC
```

**Comment by Haroon-Khel** [comment]
jdk11 openj9 on build-osuosl-aix71-ppc64-1
https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk11u/job/jdk11u-aix-ppc64-openj9/933/console

**Comment by aixtools** [comment]
> 
> 
> build-osuosl-aix71-ppc64-1 didnt even have four out of the six symlinks
> 
> ```
> /usr/bin/xlc++
> /usr/bin/xlc_r
> /usr/bin/xlC_r
> /usr/bin/gxlC
> ```

Like you said - these symlinks should not be needed.

**Comment by Haroon-Khel** [comment]
Running a jdk11 hotspot job instead
https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk11u/job/jdk11u-aix-ppc64-hotspot/871/console
The above openj9 job failed due to https://github.com/AdoptOpenJDK/openjdk-build/issues/2492. Can be fixed by running the aix `cmake` rolls. Will do so after this testing

**Comment by Haroon-Khel** [comment]
The hotspot job didnt seem to have trouble finding xlc
```
16:09:05  configure: Using default toolchain xlc (IBM XL C/C++)
16:09:05  checking for xlc_r... /opt/IBM/xlC/13.1.3/bin/xlc_r
16:09:05  checking resolved symbolic links for CC... /opt/IBM/xlC/13.1.3/bin/xlC
16:09:05  configure: Using xlc C compiler version 13.1.3 [IBM XL C/C++ for AIX, V13.1.3 (5725-C72, 5765-J07) Version: 13.01.0003.0000]
16:09:05  checking whether the C compiler works... yes
16:09:05  checking for C compiler default output file name... a.out
16:09:05  checking for suffix of executables... 
16:09:05  checking whether we are cross compiling... no
16:09:05  checking for suffix of object files... o
16:09:05  checking whether we are using the GNU C compiler... no
16:09:05  checking whether /opt/IBM/xlC/13.1.3/bin/xlc_r accepts -g... yes
16:09:05  checking for /opt/IBM/xlC/13.1.3/bin/xlc_r option to accept ISO C89... none needed
16:09:05  checking for xlC_r... /opt/IBM/xlC/13.1.3/bin/xlC_r
16:09:05  checking resolved symbolic links for CXX... /opt/IBM/xlC/13.1.3/bin/xlC
16:09:05  configure: Using xlc C++ compiler version 13.1.3 [IBM XL C/C++ for AIX, V13.1.3 (5725-C72, 5765-J07) Version: 13.01.0003.0000]
16:09:05  checking whether we are using the GNU C++ compiler... no
16:09:05  checking whether /opt/IBM/xlC/13.1.3/bin/xlC_r accepts -g... yes
16:09:06  checking how to run the C preprocessor... /opt/IBM/xlC/13.1.3/bin/xlc_r -E
16:09:06  checking how to run the C++ preprocessor... /opt/IBM/xlC/13.1.3/bin/xlC_r -E
```

**Review by Haroon-Khel — APPROVED** [review]
The jdk 11 hotspot job suceeded despite the removal of the symlinks. LGTM

**Comment by Haroon-Khel** [comment]
@sxa ping for re-review

File: ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml (hunk 3 above)
Comment (aixtools) [reply ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/xlc_v13/tasks/main.yml]: Adding a comment:

As an example:
osuosl build-{1,2} seem to pre-date this change to the use of symbolic links (build-1 has none, build-2 has 2)
```
root@nim.bak:[/export/nim/resources]for i in 1 2 3 4 5 6^Jdo ^Jprint ojdk0$i^Jssh ojdk0$i.bak ls -l /usr/bin/xl*
> done
ojdk01
ls_64: cannot access '/usr/bin/xl*': A file or directory in the path name does not exist.
ojdk02
lrwxrwxrwx 1 root system 27 Jul  2  2020 /usr/bin/xlC -> /opt/IBM/xlC/13.1.3/bin/xlC
lrwxrwxrwx 1 root system 27 Jul  2  2020 /usr/bin/xlc -> /opt/IBM/xlc/13.1.3/bin/xlc
ojdk03
lrwxrwxrwx    1 root     system           27 Jul 29 2019  /usr/bin/xlC -> /opt/IBM/xlC/13.1.3/bin/xlC
lrwxrwxrwx    1 root     system           29 Jul 29 2019  /usr/bin/xlC_r -> /opt/IBM/xlC/13.1.3/bin/xlc_r
lrwxrwxrwx    1 root     system           27 Jul 29 2019  /usr/bin/xlc -> /opt/IBM/xlC/13.1.3/bin/xlc
lrwxrwxrwx    1 root     system           29 Jul 29 2019  /usr/bin/xlc++ -> /opt/IBM/xlC/13.1.3/bin/xlc++
lrwxrwxrwx    1 root     system           29 Jul 29 2019  /usr/bin/xlc_r -> /opt/IBM/xlC/13.1.3/bin/xlc_r
ojdk04
lrwxrwxrwx    1 root     system           27 Jan 28 2020  /usr/bin/xlC -> /opt/IBM/xlC/13.1.3/bin/xlC
lrwxrwxrwx    1 root     system           29 Jan 28 2020  /usr/bin/xlC_r -> /opt/IBM/xlC/13.1.3/bin/xlc_r
lrwxrwxrwx    1 root     system           27 Jan 28 2020  /usr/bin/xlc -> /opt/IBM/xlC/13.1.3/bin/xlc
lrwxrwxrwx    1 root     system           29 Jan 28 2020  /usr/bin/xlc++ -> /opt/IBM/xlC/13.1.3/bin/xlc++
lrwxrwxrwx    1 root     system           29 Jan 28 2020  /usr/bin/xlc_r -> /opt/IBM/xlC/13.1.3/bin/xlc_r
ojdk05
lrwxrwxrwx 1 root system 27 Sep 22 13:36 /usr/bin/xlC -> /opt/IBM/xlC/13.1.3/bin/xlC
lrwxrwxrwx 1 root system 29 Sep 22 13:37 /usr/bin/xlC_r -> /opt/IBM/xlC/13.1.3/bin/xlc_r
lrwxrwxrwx 1 root system 27 Sep 22 13:36 /usr/bin/xlc -> /opt/IBM/xlC/13.1.3/bin/xlc
lrwxrwxrwx 1 root system 29 Sep 22 13:36 /usr/bin/xlc++ -> /opt/IBM/xlC/13.1.3/bin/xlc++
lrwxrwxrwx 1 root system 29 Sep 22 13:37 /usr/bin/xlc_r -> /opt/IBM/xlC/13.1.3/bin/xlc_r
ojdk06
Warning: Permanently added 'ojdk06.bak' (ECDSA) to the list of known hosts.
lrwxrwxrwx 1 root system 27 Mar  8 19:54 /usr/bin/xlC -> /opt/IBM/xlC/13.1.3/bin/xlC
lrwxrwxrwx 1 root system 29 Mar  8 19:54 /usr/bin/xlC_r -> /opt/IBM/xlC/13.1.3/bin/xlc_r
lrwxrwxrwx 1 root system 27 Mar  8 19:54 /usr/bin/xlc -> /opt/IBM/xlC/13.1.3/bin/xlc
lrwxrwxrwx 1 root system 29 Mar  8 19:54 /usr/bin/xlc++ -> /opt/IBM/xlC/13.1.3/bin/xlc++
lrwxrwxrwx 1 root system 29 Mar  8 19:54 /usr/bin/xlc_r -> /opt/IBM/xlC/13.1.3/bin/xlc_r
```
* As far as PATH definition goes - all include `/opt/IBM/xlC/13.1.3/bin` in their PATH - but again - not consistent:
```
root@nim.bak:[/export/nim/resources]for i in 1 2 3 4 5 6^Jdo ^Jprint ojdk0$i^Jssh ojdk0$i.bak grep PATH= /etc/environment | grep -i xlc^Jdone
ojdk01
PATH=/opt/freeware/bin:/opt/IBM/xlC/13.1.3/bin:/usr/bin:/etc:/usr/sbin:/usr/ucb:/usr/bin/X11:/sbin:/usr/java6/jre/bin:/usr/java6/bin
ojdk02
PATH=/opt/freeware/bin:/opt/IBM/xlC/13.1.3/bin:/usr/bin:/etc:/usr/sbin:/usr/ucb:/usr/bin/X11:/sbin:/usr/java6/jre/bin:/usr/java6/bin
ojdk03
#PATH=/opt/freeware/bin:/opt/IBM/xlC/13.1.3/bin:/usr/bin:/etc:/usr/sbin:/usr/ucb:/usr/bin/X11:/sbin:/usr/java7_64/jre/bin:/usr/java7_64/bin
PATH=/opt/IBM/xlC/13.1.3/bin:/usr/bin:/etc:/usr/sbin:/usr/ucb:/usr/bin/X11:/sbin:/usr/java7_64/jre/bin:/usr/java7_64/bin
ojdk04
PATH=/opt/IBM/xlC/13.1.3/bin:/usr/bin:/etc:/usr/sbin:/usr/ucb:/usr/bin/X11:/sbin:/usr/java7_64/jre/bin:/usr/java7_64/bin
ojdk05
PATH=/opt/freeware/bin:/opt/IBM/xlC/13.1.3/bin:/usr/bin:/etc:/usr/sbin:/usr/ucb:/usr/bin/X11:/sbin:/usr/java6/jre/bin:/usr/java6/bin
ojdk06
PATH=/opt/freeware/bin:/opt/IBM/xlC/13.1.3/bin:/usr/bin:/etc:/usr/sbin:/usr/ucb:/usr/bin/X11:/sbin:/usr/java6/jre/bin:/usr/java6/bin:/opt/bin
```

**Review by sxa — COMMENTED** [review]
@Haroon-Khel Have you tried running any of the tests suites - some of them do native code compilations so would be good to know that they don't break without the symlinks being present.

**Comment by Haroon-Khel** [comment]
@sxa The jdk11 hotspot build job posted above included a openjdk sanity test, which ran without error. Ive kicked off an extended system test, jdk 11 hotspot on build-osuosl-aix71-ppc64-1 (links removed)
https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/7539/console

**Comment by sxa** [comment]
> @sxa The jdk11 hotspot build job posted above included a openjdk sanity test, which ran without error. Ive kicked off an extended system test, jdk 11 hotspot on build-osuosl-aix71-ppc64-1 (links removed)
> https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/7539/console

The sanity test ran on https://ci.adoptopenjdk.net/computer/build-osuosl-aix71-ppc64-1/ which has the symlinks in place, so that's not a valid test for the scenario I described. I'm not sure if the system tests have native tests. I've queued up https://ci.adoptopenjdk.net/job/Test_openjdk11_hs_sanity.openjdk_ppc64_aix/346/ as a re-run of your [344](https://ci.adoptopenjdk.net/job/Test_openjdk11_hs_sanity.openjdk_ppc64_aix/344/) on [the same machine](https://ci.adoptopenjdk.net/computer/build-osuosl-aix71-ppc64-1/) as your latest Grinder which we can use to verify . Haveing said that, on that machine it looks like `/opt/IBM/xlC/13.1.3/bin` is in `/etc/environment` which I was slightly surprised by, so symlink or otherwise it's going to get picked up. Is that the case for new machines configured via the playbooks?

**Comment by Haroon-Khel** [comment]
Rerunning the sanity test on build-osuosl-aix71-ppc64-1 (which doesnt have the symlinks in place) as @sxa's previous test job failed on an unrelated error.

>Having said that, on that machine it looks like /opt/IBM/xlC/13.1.3/bin is in /etc/environment which I was slightly surprised by, so symlink or otherwise it's going to get picked up. Is that the case for new machines configured via the playbooks?

Yes, as of https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/3a5f6c1d19004209f7f7506276e9116cc50effe0/ansible/playbooks/AdoptOpenJDK_AIX_Playbook/main.yml#L123

**Comment by sxa** [comment]
@Haroon-Khel That doesn't look good:
```

12:17:30  STDOUT:
12:17:30  findLibrary someLibrary
12:17:30  Thread1 load
12:17:30  findLibrary awt
12:17:30  STDERR:
12:17:30  java.lang.UnsatisfiedLinkError: /home/jenkins/workspace/Test_openjdk11_hs_sanity.openjdk_ppc64_aix/openjdkbinary/j2sdk-image/lib/libawt_xawt.so: /home/jenkins/workspace/Test_openjdk11_hs_sanity.openjdk_ppc64_aix/openjdkbinary/j2sdk-image/lib/libawt_xawt.so, LIBPATH=/home/jenkins/workspace/Test_openjdk11_hs_sanity.openjdk_ppc64_aix/openjdkbinary/j2sdk-image/lib/server:/home/jenkins/workspace/Test_openjdk11_hs_sanity.openjdk_ppc64_aix/openjdkbinary/j2sdk-image/lib:/home/jenkins/workspace/Test_openjdk11_hs_sanity.openjdk_ppc64_aix/openjdkbinary/j2sdk-image/lib/jli:/home/jenkins/workspace/Test_openjdk11_hs_sanity.openjdk_ppc64_aix/openjdkbinary/j2sdk-image/../lib, LD_LIBRARY_PATH= : 	0509-022 Cannot load module /home/jenkins/workspace/Test_openjdk11_hs_sanity.openjdk_ppc64_aix/openjdkbinary/j2sdk-image/lib/libawt_xawt.so.
12:17:30  	0509-150   Dependent module /opt/freeware/lib/libXrender.a(shr_64.o) could not be loaded.
12:17:30  	0509-152   Member shr_64.o is not found in archive 
```

**Comment by Haroon-Khel** [comment]
[Rerunning](https://ci.adoptopenjdk.net/job/Test_openjdk11_hs_sanity.openjdk_ppc64_aix/351/console) on build-osuosl-aix71-ppc64-1 (on which the symlinks are not present)

**Comment by Haroon-Khel** [comment]
```
image/lib/jli:/home/jenkins/workspace/Test_openjdk11_hs_sanity.openjdk_ppc64_aix/openjdkbinary/j2sdk-image/../lib, LD_LIBRARY_PATH= : 	0509-022 Cannot load module /home/jenkins/workspace/Test_openjdk11_hs_sanity.openjdk_ppc64_aix/openjdkbinary/j2sdk-image/lib/libawt_xawt.so.
14:14:32  	0509-150   Dependent module /opt/freeware/lib/libXrender.a(shr_64.o) could not be loaded.
14:14:32  	0509-152   Member shr_64.o is not found in archive 
```

**Comment by Haroon-Khel** [comment]
>I wonder if its to do with https://github.com/AdoptOpenJDK/openjdk-build/blob/f3573132f91a21872583bc05e181668f313b50b6/build-farm/platform-specific-configurations/aix.sh#L33

Probably not since there isnt a duplicate of libXrender.a in `/opt/freeware/lib/pthread/ppc64`

/opt/freeware/lib/libXrender.a is present on build-osuosl-aix71-ppc64-1 and -2

**Comment by sxa** [comment]
> Same error
> 
> ```
> image/lib/jli:/home/jenkins/workspace/Test_openjdk11_hs_sanity.openjdk_ppc64_aix/openjdkbinary/j2sdk-image/../lib, LD_LIBRARY_PATH= : 	0509-022 Cannot load module /home/jenkins/workspace/Test_openjdk11_hs_sanity.openjdk_ppc64_aix/openjdkbinary/j2sdk-image/lib/libawt_xawt.so.
> 14:14:32  	0509-150   Dependent module /opt/freeware/lib/libXrender.a(shr_64.o) could not be loaded.
> 14:14:32  	0509-152   Member shr_64.o is not found in archive 
> ```
> 
> I wonder if its to do with https://github.com/AdoptOpenJDK/openjdk-build/blob/f3573132f91a21872583bc05e181668f313b50b6/build-farm/platform-specific-configurations/aix.sh#L33

That file is only explicitly called during in builds and not tests so while it does affect what's encoded into the java binaries (You can use `dump -Xany -H java` to see the library search paths encoded inside it)

Suggest trying (a) on different machines in Grinder and (b) outside jenkins to see if you can replicate it which will aid debugging

**Comment by Haroon-Khel** [comment]
[Rerunning](https://ci.adoptopenjdk.net/job/Grinder/7886/console) on test-osuosl-aix71-ppc64-1 (symlinks removed)

Im also running the failing tests, jdk_lang_0 and jdk_lang_1, on build-osuosl-aix71-ppc64-1 while playing around with the LIBPATH variable (in progress)

**Comment by Haroon-Khel** [comment]
Also running an openjdk sanity test on test-osuosl-aix72-ppc64-1 which has its symlinks in place. Using as a control.
https://ci.adoptopenjdk.net/job/Grinder/7888/console

**Comment by Haroon-Khel** [comment]
>Im also running a jdk11 openjdk sanity test on build-osuosl-aix71-ppc64-1 directly, while playing around with the LIBPATH variable (in progress)

This passed. The LIBPATH variable was empty

**Comment by Haroon-Khel** [comment]
>Also running an openjdk sanity test on test-osuosl-aix72-ppc64-1 which has its symlinks in place. Using as a control.
https://ci.adoptopenjdk.net/job/Grinder/7889/console

This passed. 

I remade the symlinks on build-osuosl-aix71-ppc64-1 and -2 and am running grinders on both
[-1](https://ci.adoptopenjdk.net/job/Grinder/7952/console) [-2](https://ci.adoptopenjdk.net/job/Grinder/7950/console)

**Comment by Haroon-Khel** [comment]
Both grinders on the build-osuosl machines still failed the following tests despite the symlinks being in place
```
jdk_lang_1
jdk_util_1
```
with the same error
```
LD_LIBRARY_PATH= : 	0509-022 Cannot load module /home/jenkins/workspace/Grinder/openjdkbinary/j2sdk-image/lib/libawt_xawt.so.
13:52:59  	0509-150   Dependent module /opt/freeware/lib/libXrender.a(shr_64.o) could not be loaded.
13:52:59  	0509-152   Member shr_64.o is not found in archive
```

**Comment by sxa** [comment]
It sounds as though the `/usr/bin/xlc` symlinks are not responsible for the problems above. If that's the case then we should consider the failure to be non-blocking for this PR - I will split out another issue to cover the failures above.

**Review by sxa — APPROVED** [review]
Approving since I don't believe the test failures are due to this change.

**Comment by Haroon-Khel** [comment]
> Approving since I don't believe the test failures are due to this change.

Can confirm. test-ibm-aix71-ppc64-2 is experiencing the same issue and its symlinks have always been there
https://ci.adoptopenjdk.net/job/Grinder/7959/consoleFull

**Review by Haroon-Khel — APPROVED** [review]
